js/ui/screenShield.js

```diff
@@ -582,7 +582,7 @@ var ScreenShield = new Lang.Class({
         this._syncInhibitor();
 
         Main.paygManager.connect('code-expired', () => { this.lock(true) });
-        Main.paygManager.connect('expiry-time-changed', () => {
+        Main.paygManager.connect('success-message-shown', () => {
```

Comment:
This will mean that any change to the expiry time that doesn't go through the lock screen will be ignored. Try provisioning a system with Endless PAYG, adding some credit, signing in, then running `eos-payg-ctl clear-code`. Does the screen lock?

Reply:
Actually it does - on the js/ui/payg.js, the tray applet can capture the signal for unlocking the screen. 

I'm not really confortable with the signal names, because here is 

code-expired -> No more time available
expiry-time-changed -> Code successfully added.

I tried to keep the same wording that the tray applet and the unlockDialog uses, but it does not makes much sense for the screenshield, and it seems weird to emit two signals with the same puporse to different objects

Reply:
Hmm, I got different behaviour just now - when i tested it out, the screen locked immediatly. After putting 4 days and trying to clear that, it didn't. However, letting time run out did had the desired outcome.